src/pages/ReportAddApproverPage.tsx

```diff
@@ -0,0 +1,205 @@
+import * as Illustrations from '@components/Icon/Illustrations';
+import ScreenWrapper from '@components/ScreenWrapper';
+import SelectionList from '@components/SelectionList';
+import InviteMemberListItem from '@components/SelectionList/InviteMemberListItem';
+import Text from '@components/Text';
+import type {SelectionListApprover} from '@components/WorkspaceMembersSelectionList';
+import useCurrentUserPersonalDetails from '@hooks/useCurrentUserPersonalDetails';
+import useDebouncedState from '@hooks/useDebouncedState';
+import useLocalize from '@hooks/useLocalize';
+import useOnyx from '@hooks/useOnyx';
+import useThemeStyles from '@hooks/useThemeStyles';
+import {addReportApprover} from '@libs/actions/IOU';
+import {canUseTouchScreen} from '@libs/DeviceCapabilities';
+import Navigation from '@libs/Navigation/Navigation';
+import type {PlatformStackScreenProps} from '@libs/Navigation/PlatformStackNavigation/types';
+import type {ReportChangeApproverParamList} from '@libs/Navigation/types';
+import {getSearchValueForPhoneOrEmail, sortAlphabetically} from '@libs/OptionsListUtils';
+import {getMemberAccountIDsForWorkspace, isPolicyAdmin} from '@libs/PolicyUtils';
+import {getDisplayNameForParticipant, isMoneyRequestReport, isMoneyRequestReportPendingDeletion} from '@libs/ReportUtils';
+import tokenizedSearch from '@libs/tokenizedSearch';
+import variables from '@styles/variables';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import ROUTES from '@src/ROUTES';
+import type SCREENS from '@src/SCREENS';
+import {isEmptyObject} from '@src/types/utils/EmptyObject';
+import NotFoundPage from './ErrorPage/NotFoundPage';
+import type {WithReportOrNotFoundProps} from './home/report/withReportOrNotFound';
+import withReportOrNotFound from './home/report/withReportOrNotFound';
+
+type ReportAddApproverPageProps = WithReportOrNotFoundProps & PlatformStackScreenProps<ReportChangeApproverParamList, typeof SCREENS.REPORT_CHANGE_APPROVER.ADD_APPROVER>;
+
+function ReportAddApproverPage({report, isLoadingReportData, policy}: ReportAddApproverPageProps) {
+    const styles = useThemeStyles();
+    const {translate, localeCompare} = useLocalize();
+    const [searchTerm, debouncedSearchTerm, setSearchTerm] = useDebouncedState('');
+    const [selectedApproverEmail, setSelectedApproverEmail] = useState<string | undefined>(undefined);
+    const [personalDetails] = useOnyx(ONYXKEYS.PERSONAL_DETAILS_LIST, {canBeMissing: false});
+    const [countryCode] = useOnyx(ONYXKEYS.COUNTRY_CODE, {canBeMissing: false});
+
+    const currentUserDetails = useCurrentUserPersonalDetails();
+
+    const [allApprovers, setAllApprovers] = useState<SelectionListApprover[]>([]);
+    const shouldShowTextInput = allApprovers?.length >= CONST.STANDARD_LIST_ITEM_LIMIT;
+
+    const employeeList = policy?.employeeList;
+    const sections = useMemo(() => {
+        const approvers: SelectionListApprover[] = [];
+
+        if (employeeList) {
```

Comment:
Maybe better to default `employeeList` to some value than conditionally check if it exists. I would also think that this is not possible at all so if there is no employeeList we should log something.